reporter/src/main/kotlin/reporters/fossid/FossIdReporter.kt

```diff
@@ -0,0 +1,104 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters.fossid
+
+import java.io.File
+import java.time.Duration
+
+import kotlinx.coroutines.runBlocking
+
+import org.ossreviewtoolkit.clients.fossid.FossIdRestService
+import org.ossreviewtoolkit.clients.fossid.FossIdServiceWithVersion
+import org.ossreviewtoolkit.clients.fossid.generateReport
+import org.ossreviewtoolkit.clients.fossid.model.report.ReportType
+import org.ossreviewtoolkit.clients.fossid.model.report.SelectionType
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.utils.core.OkHttpClientHelper
+import org.ossreviewtoolkit.utils.core.log
+
+class FossIdReporter : Reporter {
+    companion object {
+        /** Name of the configuration property for the server URL. */
+        const val SERVER_URL_PROPERTY = "serverUrl"
+
+        /** Name of the configuration property for the API key. */
+        const val API_KEY_PROPERTY = "apiKey"
+
+        /** Name of the configuration property for the user name. */
+        const val USER_PROPERTY = "user"
+
+        /** Name of the configuration property for the report type. Default is [ReportType.HTML_DYNAMIC]. */
+        const val REPORT_TYPE_PROPERTY = "reportType"
+
+        /**
+         * Name of the configuration property for the selection type.
+         * Default is [SelectionType.INCLUDE_ALL_LICENSES].
+         */
+        const val SELECTION_TYPE_PROPERTY = "selectionType"
+
+        /**
+         * Name of key in [ScanResult.additionalData] containing the scancode.
+         * TODO: It should be unified with [FossId.SCAN_CODE_KEY], without creating a dependency between scanner and
+         * reporter.
+         */
+        const val SCAN_CODE_KEY = "scancode"
+    }
+
+    override val reporterName = "FossId"
+
+    internal fun createService(serverUrl: String): FossIdRestService {
+        val client = OkHttpClientHelper.buildClient {
+            readTimeout(Duration.ofSeconds(60))
+        }
+        return FossIdRestService.create(serverUrl, client).let { FossIdServiceWithVersion.instance(it) }.also {
+            if (it.version.isEmpty()) {
+                log.warn { "The FossID server is running an unknown version." }
+            } else {
+                log.info { "The FossID server is running version ${it.version}." }
+            }
+        }
+    }
```

Comment:
This code is only slightly different from the code in `scanner/src/main/kotlin/scanners/fossid/FossIdConfig.kt`. Can we align / extract it? E.g. by creating a client with increased timeout already in

https://github.com/oss-review-toolkit/ort/blob/c8c18090f49018e55b2095fd1be202eac3fe95bf/clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt#L125

if no other client is given, and also doing the logging already over there?